### PR TITLE
Gt4Py SIR Test

### LINKED ISSUE #704: Ensure GT4Py examples are compatible with dawn backends (generates valid SIR)
Ensure that valid SIR can be generated from GT4Py examples (e.g., horizontal difference, tridiagonal solve, Burgers') and can be compiled by Dawn backends. Coordinate with Enrique.

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Technical Description

 Ensure that GT4Py examples are compatible with dawn backends (generates valid SIR). This PR adds infrastructure and test to confirm that GT4Py example stencils can be converted to SIR and used to generate code. Supporting changes to dawn4py were also implemented. The following examples were tested:

* Tridiagonal Solve
* Horizontal Difference
* Burgers Demo
* Coriolis Stencil

### Resolves / Enhances

Resolves #704

### CODE REVIEW

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/SIR/TestSIRFrontend.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+class TestSIRFrontend : public ::testing::Test {
+protected:
+  void runTest(const std::string& irFilename, const std::string frontend = "gt4py") {
+    std::string refPath = irFilename + "_ref.sir";
+    auto refSIR = SIRSerializer::deserialize(refPath);
+
+    std::string testPath = irFilename + "_" + frontend + ".sir";
+    auto testSIR = SIRSerializer::deserialize(testPath);
+
+    ASSERT_TRUE(*refSIR == *testSIR);
```
Comment (jdahm) [inline dawn/test/unit-test/dawn/SIR/TestSIRFrontend.cpp]: Nice!

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Pull request merged**